### PR TITLE
Show deprecated collection status

### LINKED ISSUE #93: Allow to declare collections as unmaintained, ... for Ansible docsite builds
That should use https://github.com/ansible-community/ansible-build-data/blob/main/8/collection-meta.yaml somehow.

### PR DESCRIPTION
**Pull request by felixfontein**
Fixes #93.

- [x] Depend on antsibull-core >= 3.1.0 and pydantic >= 2.0.0
- [x] Load collection meta
- [x] Indicate deprecated collections in the ansible devel and sable docsites (fixes #93)

### CODE REVIEW

**Comment by felixfontein** [comment]
We should also get rid of the v1 usage, but that will be a larger effort, especially for the docs schemas and the linting. I think we need a generic solution for https://github.com/ansible-community/antsibull-core/blob/7a6d701a6070f13496fe68f23514784532c2938a/src/antsibull_core/collection_meta.py#L143-L151; we really don't want to keep track of such a list of classes (and in particular its order) manually for the docs schemas.

But that's something for follow-up PRs, for now let's stick to removing the v1 support of the compatibility layer.

**Comment by felixfontein** [comment]
I generated the stable docs RST files for 10.4.0, here are some excerpts:

List of all collections (collections/index.rst):
```rst
Collection Index
================

These are the collections with docs hosted on `docs.ansible.com <https://docs.ansible.com/>`_.

* :ref:`amazon.aws <plugins_in_amazon.aws>`
...
* :ref:`fortinet.fortios <plugins_in_fortinet.fortios>`
* :ref:`frr.frr <plugins_in_frr.frr>` :ansdeprecatedmarker:`{"version": 11}`
* :ref:`google.cloud <plugins_in_google.cloud>`
...
* :ref:`ibm.qradar <plugins_in_ibm.qradar>`
* :ref:`ibm.spectrum_virtualize <plugins_in_ibm.spectrum_virtualize>` :ansdeprecatedmarker:`{"version": null}`
* :ref:`ibm.storage_virtualize <plugins_in_ibm.storage_virtualize>`
...
* :ref:`inspur.ispim <plugins_in_inspur.ispim>`
* :ref:`inspur.sm <plugins_in_inspur.sm>` :ansdeprecatedmarker:`{"version": 11}`
* :ref:`junipernetworks.junos <plugins_in_junipernetworks.junos>`
...
```
Deprecation notes for collections:
```rst
Frr.Frr
=======

Collection version 2.0.2

.. note::
    The frr.frr collection has been deprecated and will be removed from Ansible 11.
    See the \ `discussion thread <https://forum.ansible.com/t/6243>`__ for more information.

Ibm.Spectrum_Virtualize
=======================

Collection version 2.0.0

.. note::
    The ibm.spectrum\_virtualize collection has been renamed to :ref:`ibm.storage\_virtualize <plugins_in_ibm.storage_virtualize>` and will eventually be removed from Ansible.
    If you use content from ibm.spectrum\_virtualize, please update FQCNs in your playbooks and roles!
    When creating new playbooks or roles, directly use content from ibm.storage\_virtualize instead.

Inspur.Sm
=========

Collection version 2.3.0

.. note::
    The inspur.sm collection is considered unmaintained and will be removed from Ansible 11.
    See the \ `discussion thread <https://forum.ansible.com/t/2854>`__ for more information.

Netapp.Storagegrid
==================

Collection version 21.12.0

.. note::
    The netapp.storagegrid collection is considered unmaintained and will be removed from Ansible 11.
    See the \ `discussion thread <https://forum.ansible.com/t/2811>`__ for more information.

Openvswitch.Openvswitch
=======================

Collection version 2.1.1

.. note::
    The openvswitch.openvswitch collection has been deprecated and will be removed from Ansible 11.
    See the \ `discussion thread <https://forum.ansible.com/t/6245>`__ for more information.
```

**Review by oraNod — APPROVED** [review]
From what I can see this looks good. Nice work @felixfontein

File: src/antsibull_docs/cli/doc_commands/_build.py [hunk 1]
```diff
@@ -165,6 +178,132 @@ def _register_extra_docs(
...
+    new_name: str,
+    redirect_replacement_major_version: int,
+    current_major_version: int | None,
+) -> str:
+    if (
+        current_major_version is not None
+        and current_major_version == redirect_replacement_major_version
+    ):
+        return (
+            f"The content of {collection} have been replaced"
```
Comment (samccann) [inline src/antsibull_docs/cli/doc_commands/_build.py:192]:
```suggestion
            f"The content of {collection} has been replaced"
```

Comment (samccann) [reply src/antsibull_docs/cli/doc_commands/_build.py:192]: We could go two ways here:
"The contents of foo have been"
or
"The content of foo has been"

Given that other text uses 'content' frequently, I recommend the suggested edit here and on other lines like this.

File: src/antsibull_docs/cli/doc_commands/_build.py [hunk 2]
```diff
@@ -165,6 +178,132 @@ def _register_extra_docs(
...
+            " please update FQCNs in your playbooks and roles!"
+        )
+        sentences.append(
+            "When creating new playbooks or roles,"
+            f" directly use content from {removal.new_name} instead."
+        )
+    if removal.reason == "guidelines-violation":
+        sentences.append(
+            f"The {collection} collection {removed_text}"
+            " due to violations of the Ansible inclusion."
```
Comment (samccann) [inline src/antsibull_docs/cli/doc_commands/_build.py:258]:
```suggestion
            " due to violations of the Ansible inclusion requirements."
```

**Review by samccann — COMMENTED** [review]
just some wording nits.. .thanks for all this!

File: src/antsibull_docs/cli/doc_commands/_build.py (hunk 1 above)
Comment (felixfontein) [reply src/antsibull_docs/cli/doc_commands/_build.py]: I changed it to `content ... has been` in 167496b14485edb44cc585dbc865bf01e7a6d788.

**Comment by felixfontein** [comment]
@samccann @oraNod thanks for your contribution!

**Pull request merged**